tests/simulator_test.cpp

```diff
@@ -2,82 +2,217 @@
-  ASSERT_FALSE(performHitAtEnergyCollectors(tiltedFromXY, &hitData));
-
-  Ray straightUpMoved(
-      Vec3(2 * manager.collectorRadius(), 2 * manager.collectorRadius(), 0),
-      Vec3(0, 0, 1));
-  ASSERT_FALSE(performHitAtEnergyCollectors(straightUpMoved, &hitData));
-} */
\ No newline at end of file
+  ASSERT_EQ(performHitCollector(energyCollectors, straightUp, &hitData), true);
+
+  const float collectorPositionRadius = 4;
+  const float collectorAngle = 2 * kPi / (numCollectors - 1);
+  const float refCollectorRadius =
+      collectorPositionRadius * std::sqrt(2 - 2 * std::cos(collectorAngle));
+
+  Vec3 collisionPointStraightUp =
+      Vec3(0, 0, collectorPositionRadius - refCollectorRadius);
+  ASSERT_EQ(hitData.collisionPoint(), collisionPointStraightUp);
+
+  Ray straightDown(kVecZero, -kVecUp);
+  ASSERT_FALSE(performHitCollector(energyCollectors, straightDown, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+
+  Ray alongX(kVecZero, kVecX);
+  ASSERT_TRUE(performHitCollector(energyCollectors, alongX, &hitData));
+
+  ASSERT_EQ(alongX.at(collectorPositionRadius - refCollectorRadius),
+            hitData.collisionPoint());
+
+  Ray alongY(kVecZero, kVecY);
+  ASSERT_TRUE(performHitCollector(energyCollectors, alongY, &hitData));
+
+  ASSERT_EQ(alongY.at(collectorPositionRadius - refCollectorRadius),
+            hitData.collisionPoint());
+
+  Ray atSixty(kVecZero, Vec3(std::cos(deg2rad(60)), 0, std::sin(deg2rad(60))));
+  ASSERT_TRUE(performHitCollector(energyCollectors, atSixty, &hitData));
+
+  Ray atSixtyOther(kVecZero,
+                   Vec3(-std::cos(deg2rad(60)), 0, std::sin(deg2rad(60))));
+  ASSERT_TRUE(performHitCollector(energyCollectors, atSixtyOther, &hitData));
+
+  Ray atSixtyXY(kVecZero, Vec3(std::cos(deg2rad(60)), std::cos(deg2rad(60)),
+                               std::sin(deg2rad(60))));
+  ASSERT_FALSE(performHitCollector(energyCollectors, atSixtyXY, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+
+  Ray atSixtyXYOther(kVecZero,
+                     Vec3(-std::cos(deg2rad(60)), -std::cos(deg2rad(60)),
+                          std::sin(deg2rad(60))));
+  ASSERT_FALSE(performHitCollector(energyCollectors, atSixtyXYOther, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+}
+
+TEST_F(EnergyCollectorTest, EvenNumOfEnergyCollectorTest) {
+  const FakeModel nonEmptyModel(false);
+
+  const int numCollectors = 36;
+  auto energyCollectors = buildCollectors(nonEmptyModel, numCollectors);
+  ASSERT_EQ(energyCollectors.size(), numCollectors);
+
+  Ray straightUp(kVecZero, kVecUp);
+  RayHitData hitData;
+  ASSERT_TRUE(performHitCollector(energyCollectors, straightUp, &hitData));
+
+  const float collectorPositionRadius = 4;
+  const float collectorAngle = 2 * kPi / (numCollectors - 2);
+  const float refCollectorRadius =
+      collectorPositionRadius * std::sqrt(2 - 2 * std::cos(collectorAngle));
+
+  Vec3 OriginOfPenultimateCollector =
```

Comment:
What if the order of placing collectors was different? In the current setup you are leaking the implementation details to the test which means that you are not actually testing much, besides that code in the test matches code in the impl.

Reply:
if the order of placing collector would be different, the test would be completly usless and nanoying. But I really dont know how to get collision point betwen to collectors for the enev number of collectors (see other comment). I decided to do this test knowing how energy collectors are placed -> do you thinkg that I should write comment about it? ;-)

Reply:
No comment. Go back to the previous impl. and do not rely on this. This will make your tests/code brittle.